dbms/src/Functions/FunctionBase64Conversion.h

```diff
@@ -0,0 +1,150 @@
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/GatherUtils/Algorithms.h>
+#include <IO/WriteHelpers.h>
+#include <Common/config.h>
+
+#if USE_BASE64
+#include <libbase64.h>
+
+
+namespace DB
+{
+using namespace GatherUtils;
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int INCORRECT_DATA;
+}
+
+struct Base64Encode
+{
+    static constexpr auto name = "base64Encode";
+    static constexpr auto buffer_size_multiplier = 5.0 / 3.0;
```

Comment:
`f`
is transformed to
`Zg==`

In character buffer:
`f\0`
`Zg==\0`
2 bytes are transformed to 5 bytes in worst case.

This means that the ratio should be 5/2.

Reply:
Why using so rough calculations with ceil and floats? That will allocate twice more memory than needed,

Actually it's trivial to calculate exact size by simple formula `( x / 3 ) * 4` using integer division and adding one  exctra chunk size  + 1 extra byte for \0. Something like that. 
```cpp
size_t calculateBufferSizeForEncode(size_t  size) {
    return ( size / 3 ) * 4 + 5;  /* 5 = length of one 4 byte chunk + null-byte string terminator */
}

size_t calculateBufferSizeForDecode(size_t  size) {
    return ( size / 4 ) * 3 + 4;  /* 5 = length of one 3 byte chunk + null-byte string terminator */
}
```

Reply:
This calculation is for buffer size where all resulting strings will be places together.

Reply:
Yes, this is far better!
PS. 
> extra 4 bytes of alignment

extra 3 bytes of alignment.

Reply:
> extra 3 bytes of alignment.

Sorry, alignment here is not the proper term here so it lead to some misunderstanding.
In worst case for every sting one extra chunk of dara (4 bytes) would be needed (just simulate ceil instead of floor for every string). I'll change the comment right now.
(example: 2 one byte strings. Size=4, count=2. Integer division will return 0.)

Reply:
Implemented according to your comments. Thanks!